coffeecard/CoffeeCard.Library/Services/v2/MenuItemService.cs

```diff
@@ -0,0 +1,103 @@
+using System.Collections.Generic;
+using System.Data;
+using System.Linq;
+using System.Threading.Tasks;
+using CoffeeCard.Common.Errors;
+using CoffeeCard.Library.Persistence;
+using CoffeeCard.Models.DataTransferObjects.v2.Product;
+using CoffeeCard.Models.DataTransferObjects.v2.Products;
+using CoffeeCard.Models.Entities;
+using Microsoft.EntityFrameworkCore;
+using Serilog;
+
+namespace CoffeeCard.Library.Services.v2
+{
+    public sealed class MenuItemService : IMenuItemService
+    {
+        private readonly CoffeeCardContext _context;
+
+        public MenuItemService(CoffeeCardContext context)
+        {
+            _context = context;
+        }
+
+        public async Task<IEnumerable<MenuItemResponse>> GetAllMenuItemsAsync()
+        {
+            return await _context.MenuItems
+                .OrderBy(p => p.Id)
+                .Select(p => new MenuItemResponse
+                {
+                    Id = p.Id,
+                    Name = p.Name
+                })
+                .ToListAsync();
+        }
+
+
+        private async Task<bool> CheckMenuItemUniquenessAsync(string name, int id = 0)
```

Comment:
I'm not sure I can see in what scenario the `id` would be `0`. Suggest to check by the name of a menu item

Reply:
Hmm, I think I somewhat over-engineered this part. My thought was along the lines: If I want to update a menu item, I want to check that I am not saving it as an existing name for a different menu item - but it should not throw an exception if I try to "save" it with the same name. This would make sense if a menu item has more properties - but since we only have the name property, we do not need it. We can just handle it in the frontend (don't save if we do not make changes), and then change this method in the future when (and if) we need it.

If you agree I'll remove the id part of this method, otherwise I suggest making an extra method such that we have a method checking for uniqueness for new menu item (does not take id into account) and for updating menu items (taking id into account).

Reply:
I don't see us hit that scenario at this time so suggest to remove the id